### PR TITLE
transition api methods to interfaces

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

replaces the new api client methods with interfaces, to prepare for testing funcions with mocks instead of integration.

## Does this PR need a docs update or release note?

- [x] :no_entry: No 

## Type of change

- [x] :robot: Test

## Issue(s)

* #1967

## Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/exchange/api/api.go [hunk 1]
```diff
@@ -87,6 +87,18 @@ func newService(creds account.M365Config) (*graph.Service, error) {
...
+}
+
+func (c Client) Events() Events {
+	return Events{c}
+}
+
+func (c Client) Mail() Mail {
+	return Mail{c}
+}
+
```
Comment (dadams39) [inline src/internal/connector/exchange/api/api.go:101]: This appears a little circular. The creds and service associated are available within the client already. As this is being merged into main. A small comment about about the purpose of these calls would be helpful It's also unclear why there wouldn't be similar calls for onedrive or sharepoint.

File: src/internal/connector/exchange/event_calendar_cache.go [hunk 2]
```diff
@@ -14,7 +13,7 @@ var _ graph.ContainerResolver = &eventCalendarCache{}
 
 type eventCalendarCache struct {
 	*containerResolver
-	ac     api.Client
+	enumer enumerateContainerser
```
Comment (dadams39) [inline src/internal/connector/exchange/event_calendar_cache.go:16]: Perhaps ContainersEnumerater ?

File: src/internal/connector/exchange/exchange_data_collection.go [hunk 3]
```diff
@@ -142,11 +142,11 @@ func (col *Collection) Items() <-chan data.Stream {
...
 	switch category {
 	case path.ContactsCategory:
-		return ac.RetrieveContactDataForUser, serializeAndStreamContact
+		return ac.Contacts().RetrieveContactDataForUser, serializeAndStreamContact
 	case path.EventsCategory:
-		return ac.RetrieveEventDataForUser, serializeAndStreamEvent
+		return ac.Events().RetrieveEventDataForUser, serializeAndStreamEvent
 	case path.EmailCategory:
-		return ac.RetrieveMessageDataForUser, serializeAndStreamMessage
+		return ac.Mail().RetrieveMessageDataForUser, serializeAndStreamMessage
```
Comment (dadams39) [inline src/internal/connector/exchange/exchange_data_collection.go:149]: I imagine a few PRs away, there is another one where this becomes an interface and it is simply `RetrieveDataForUser`?

File: src/internal/connector/exchange/folder_resolver_test.go [hunk 4]
```diff
@@ -50,14 +50,15 @@ func (suite *CacheResolverSuite) TestPopulate() {
...
+			enumer: ac.Events(),
 		}
 	}
 
 	contactFunc := func(t *testing.T) graph.ContainerResolver {
 		return &contactFolderCache{
 			userID: tester.M365UserID(t),
-			ac:     ac,
+			enumer: ac.Contacts(),
+			getter: ac.Contacts(),
```
Comment (dadams39) [inline src/internal/connector/exchange/folder_resolver_test.go:61]: Is this a typo? These are both the same.

File: src/internal/connector/exchange/service_iterators.go [hunk 5]
```diff
@@ -16,6 +15,13 @@ import (
 	"github.com/alcionai/corso/src/pkg/selectors"
 )
 
+type addedAndRemovedItemIDsGetter interface {
+	GetAddedAndRemovedItemIDs(
```
Comment (dadams39) [inline src/internal/connector/exchange/service_iterators.go:19]: `GetItemIDsAddedAndRemoved`

File: src/internal/connector/exchange/service_iterators.go [hunk 6]
```diff
@@ -89,7 +93,7 @@ func filterContainersAndFillCollections(
 			}
 		}
 
-		added, removed, newDelta, err := getJobs(ctx, qp.ResourceOwner, cID, prevDelta)
+		added, removed, newDelta, err := getter.GetAddedAndRemovedItemIDs(ctx, qp.ResourceOwner, cID, prevDelta)
```
Comment (dadams39) [inline src/internal/connector/exchange/service_iterators.go:96]: There is a slight stutter here: `getter.Get...`. Refactor could be `getter.ItemIDsAddedAndRemoved()`

**Review by dadams39 — APPROVED** [review]
Nits: The changes / suggestions have to do with refactoring for the sake of the API. No observable security concerns.

File: src/internal/connector/exchange/api/api.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/connector/exchange/api/api.go:101]: The comment would be, succinctly, `interface compliance wrappers`.  Not much more to tell than that.

> It's also unclear why there wouldn't be similar calls for onedrive or sharepoint.

Note the package location.  onedrive and sharepoint have no api packages as of yet.

File: src/internal/connector/exchange/event_calendar_cache.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/connector/exchange/event_calendar_cache.go:16]: Yeah, i like that better.

File: src/internal/connector/exchange/exchange_data_collection.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/internal/connector/exchange/exchange_data_collection.go:149]: Most certainly!

File: src/internal/connector/exchange/service_iterators.go (hunk 5 above)
Comment (ryanfkeepers) [reply src/internal/connector/exchange/service_iterators.go:19]: Kinda sixes here.

File: src/internal/connector/exchange/service_iterators.go (hunk 6 above)
Comment (ryanfkeepers) [reply src/internal/connector/exchange/service_iterators.go:96]: If the stutter were a concern, in this case it'd be better to rename the interface than the func.  Due to golang's duck typing, there can be many interfaces that a given struct may fulfill; it's more sane to have each interface avoid stutters than to burden the source func with that worry.

**Pull request merged**